models/DataObject/ClassDefinition/Data/Extension/Relation.php

```diff
@@ -63,14 +63,27 @@ protected function getPhpDocClassString($asArray = false)
                 $class[] = '\Pimcore\Model\DataObject\AbstractObject' . $strArray;
             } elseif (is_array($classes)) {
                 foreach ($classes as $item) {
-                    $class[] = sprintf('\Pimcore\Model\DataObject\%s', ucfirst($item['classes']) . $strArray);
+                    $class[] = $this->getMappedClassName('\Pimcore\Model\DataObject\\' . ucfirst($item['classes'])) . $strArray;
                 }
             }
         }
 
         return $class;
     }
 
+    protected function getMappedClassName(string $className): string
+    {
+        try {
+            $className = \Pimcore::getContainer()->get('pimcore.model.factory')->getClassNameFor($className);
```

Comment:
With 10.6 we introduce custom document types so the namespace for the documents are not hardcoded anymore.
The pimcore.model.factory only delivers the class_overrides which won't be enough for the documents since if you add a custom document you will run into the exception.

We should use the typeloader for the documents as well to make sure also custom document types are shown. 
https://github.com/pimcore/pimcore/blob/11.x/models/Document.php#L211

We do also have to consider this for the assets for 11.x later on.